ledfx/effects/random_spotlight.py

```
@@ -0,0 +1,396 @@
+        )
+
+    def _ring_distance(self, pixel_a, pixel_b):
+        diff = abs(pixel_a - pixel_b)
+        return min(diff, self.pixel_count - diff)
+
+    def _pick_spot_center(self):
+        if self.pixel_count <= 1:
+            return 0
+
+        if not self.active_spots:
+            return random.randrange(self.pixel_count)
```

Ruff (0.15.10)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)